shell/common/shell_test_platform_view_metal.mm

```diff
@@ -61,11 +37,28 @@
     const std::shared_ptr<const fml::SyncSwitch>& is_gpu_disabled_sync_switch)
     : ShellTestPlatformView(delegate, task_runners),
       GPUSurfaceMetalDelegate(MTLRenderTargetType::kMTLTexture),
-      metal_context_(
-          CreateDarwinContext(GetSettings().enable_impeller, is_gpu_disabled_sync_switch)),
       create_vsync_waiter_(std::move(create_vsync_waiter)),
       vsync_clock_(std::move(vsync_clock)),
-      shell_test_external_view_embedder_(std::move(shell_test_external_view_embedder)) {}
+      shell_test_external_view_embedder_(std::move(shell_test_external_view_embedder)) {
+  id<MTLDevice> device = nil;
+  if (GetSettings().enable_impeller) {
+    impeller_context_ =
+        [[FlutterDarwinContextMetalImpeller alloc] init:is_gpu_disabled_sync_switch];
+    FML_CHECK(impeller_context_.context);
+    device = impeller_context_.context->GetMTLDevice();
+  } else {
+    skia_context_ = [[FlutterDarwinContextMetalSkia alloc] initWithDefaultMTLDevice];
+    FML_CHECK(skia_context_.mainContext);
+    device = skia_context_.device;
+  }
```

Comment:
In a followup patch I'll replace the separate `impeller_context_` and `skia_context_` fields with a `std::variant` since we never use both.